### PR TITLE
Fix episode parsing

### LINKED ISSUE #2171: Tv Episodes identified incorrectly in 10.4.2 and 10.4.3
**Describe the bug**
Some TV Episodes are identified incorrectly in versions after 10.4.1. 

**To Reproduce**

Some Show 25x39 - [WEBDL-720p][x264] Some Episode.mkv mis-identified as S01E25
That Other Show 1x07 [WEBDL-1080p][EAC3 5.1][x264] - Chapter 7.mkv misidentified as S01E01

**Expected behavior**

Correctly id shows based on SSxEE number, as was occuring in 10.4.1 and lower.

**System (please complete the following information):**
 - OS: Docker
 - Browser: Chrome
 - Jellyfin Version: 10.4.2 and 10.4.3
 - Installed Plugins: Fanart, Trakt, 
 - Reverse proxy: no

### PR DESCRIPTION
**Pull request by Bond-009**
**Issues**
Fixes #2171

### CODE REVIEW

**Comment by SenorSmartyPants** [comment]
Please add a season number test with the same Daily show filename. Currently JF ids that episode as 1x25 and not the correct 25x22.

**Comment by SenorSmartyPants** [comment]
https://github.com/jellyfin/jellyfin/blob/620047ab9adeea5fc0d38f51990d456b1300b968/Emby.Naming/Common/NamingOptions.cs#L317-L325

These expressions look like the correct ones to fix the tests. They are KODI naming standard.
https://kodi.wiki/view/Naming_video_files/TV_shows
I think maybe they just need to be moved before the EpisodeExpression that is currently matching these test cases.

**Comment by Bond-009** [comment]
We moved that expression before the Kodi expressions because of false positives.

**Comment by Artiume** [comment]
Not sure if we should modify it further but I do receive false positives when the name is within the name of another movie. `Thomas and Friends: The Great Race` was matched as `The Making of Thomas and Friends: The Great Race`

This is under the current regex, not this proposed regex.

File: tests/Jellyfin.Naming.Tests/TV/EpisodeNumberTests.cs [hunk 1]
```diff
@@ -6,6 +6,21 @@ namespace Jellyfin.Naming.Tests.TV
 {
     public class EpisodeNumberTests
     {
+        private readonly NamingOptions _namingOptions = new NamingOptions();
+
+        [Theory]
+        [InlineData("Watchmen (2019)/Watchmen 1x03 [WEBDL-720p][EAC3 5.1][h264][-TBS] - She Was Killed by Space Junk.mkv", 3)]
+        [InlineData("The Daily Show/The Daily Show 25x22 - [WEBDL-720p][AAC 2.0][x264] Noah Baumbach-TBS.mkv", 22)]
+        [InlineData("Castle Rock 2x01 Que el rio siga su curso [WEB-DL HULU 1080p h264 Dual DD5.1 Subs].mkv", 1)]
+        [InlineData("'After Life 1x06 Episodio 6 [WEB-DL NF 1080p h264 Dual DD 5.1 Sub].mkv", 6)]
```
Comment (dkanada) [inline tests/Jellyfin.Naming.Tests/TV/EpisodeNumberTests.cs]: Is the single quote intentional?

Comment (Bond-009) [reply tests/Jellyfin.Naming.Tests/TV/EpisodeNumberTests.cs]: Fixed

**Comment by Bond-009** [comment]
@Artiume Create an issue with the full filename

**Pull request merged**